### PR TITLE
:bug: Source Titok marketing: update schema. add unittest, update SAT, fix initialization for Audience streams

### LINKED ISSUE #9491: Source Tiktok: wrong type in campaign schema
## Environment
- **Airbyte version**:  0.32.2-alpha
- **OS Version / Instance**: Not informed.
- **Deployment**: Not informed.
- **Source Connector and version**: Tiktok 0.1.1
- **Destination Connector and version**:  Bigquery 0.3.7
- **Severity**: Very Low / **Low** / Medium / High / Critical
- **Step where error happened**: Deploy / Sync job / Setup new connection / Update connector / Upgrade Airbyte

## Current Behavior
Sync fails during normalization

## Expected Behavior
Should work

## Logs
Complete logs: [logs-106032-0.txt](https://github.com/airbytehq/airbyte/files/7866417/logs-106032-0.txt)

## Steps to Reproduce
1.
2.
3.

## Are you willing to submit a PR?

No

### LINKED ISSUE #9709: Source Tiktok: data type error for `mobile_app_id` field in ads_reports_metrics and ad_groups_reports_metrics
## Environment
- OS Version / Instance: Ubuntu 20.04
- Machine type: e2-standard-8 ( 8 CPU )
- Memory / Disk: 16Gb / 100 GB ssd
- Deployment: Docker
- Airbyte Version: 0.35.1-alpha
- Source name/version: airbyte/source-tiktok-marketing:0.1.3
- Destination name/version: airbyte/destination-bigquery:0.6.5

## Current Behavior
Two tables from tiktok_ads are getting invalid field type, those are: ads_reports_metrics ,  ad_groups_reports_metrics.

## Expected Behavior
Sync works

## Logs
Complete logs: [logs-148-2.txt](https://github.com/airbytehq/airbyte/files/7917141/logs-148-2.txt)

## Steps to Reproduce
1.
2.
3.

## Are you willing to submit a PR?

Remove this with your answer.

### LINKED ISSUE #11029: [EPIC] Source TikTok to Cloud Beta
According to [checklist](https://docs.google.com/spreadsheets/d/1L4LNVgd72nM6BjzWKpqWtiM6EVh8Cp5lkl1BKYBACAw/edit#gid=1288588270) improvements:

- MLP requirements as surfaced through [research](https://docs.google.com/document/d/1CZ-0H9BfNaP7FQrts2MnpGFkilvLwCqulhnLCrDo0mA/edit):
  - [ ] #11368

- Engineering Acceptance Criteria:
   - [x] #11372
   - [x] #11376
   - [x] #11377

- Outstanding bugs:
   - [x] https://github.com/airbytehq/airbyte/issues/9709
   - [x] https://github.com/airbytehq/airbyte/issues/9491
   - [x] https://github.com/airbytehq/airbyte/issues/6875 
   - [x] #11375
   - [x] #12107

### LINKED ISSUE #12107: Source Titok marketing: discovery method is too long (~ 40 seconds)
Problem:
discovery method is too long

Reason:
each initialization method in stream contains read method to get list of advertiser ids

Solution:
advertiser ids should be extracted only during read process.

### PR DESCRIPTION
**Pull request by midavadim**
Engineering Acceptance Criteria:

 Source Tiktok Marketing: have at least 90% unit test coverage #11372
 Source Tiktok Marketing: add missing streams in SAT #11376
 Source Tiktok marketing: add SAT tests for DAY, HOUR granularity #11377

Outstanding bugs:

 Source Tiktok: data type error for mobile_app_id field in ads_reports_metrics and ad_groups_reports_metrics #9709
 Source Tiktok: wrong type in campaign schema #9491
 Round value of source attribute with type "integer" #6875
 Source Tiktok: fix metrix for LIFETIME granularity #11375
 Source Titok marketing: discovery method is too long (~ 40 seconds) #12107

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-tiktok-marketing/acceptance-test-config.yml [hunk 1]
```diff
@@ -22,30 +25,52 @@ tests:
...
+
   basic_read:
-    - config_path: "secrets/config.json"
-      configured_catalog_path: "integration_tests/configured_catalog.json"
-      empty_streams: [ "ads" , "ads_reports", "ad_groups_reports", "campaigns_reports" ]
+    # DAY granularity:
+    - config_path: "secrets/prod_config_day.json"
+      configured_catalog_path: "integration_tests/streams_all.json"
+      timeout_seconds: 1200
+      validate_schema: false  # validation fails for audience report streams
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-tiktok-marketing/acceptance-test-config.yml:41]: Are we aware of what exactly is wrong with the validation of `Audience Report` streams?

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/schemas/audience_reports.json [hunk 2]
```diff
@@ -43,12 +43,6 @@
         "spend": {
           "type": ["null", "number"]
         },
-        "cash_spend": {
-          "type": ["null", "number"]
-        },
-        "voucher_spend": {
-          "type": ["null", "number"]
-        },
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/schemas/audience_reports.json:51]: Is there any valid reason we removed this from the schema?

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py [hunk 3]
```diff
@@ -108,6 +114,14 @@ class TiktokStream(HttpStream, ABC):
     # max value of page
     page_size = 1000
 
+    def __init__(self, **kwargs):
+        super().__init__(authenticator=kwargs.get("authenticator"))
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:118]: Do we need to call `authenticator` directly here? I believe it should be passed automatically if we have such a property inside the passing args. Please check.

**Review by tuliren — APPROVED** [review]

**Pull request merged**